app/governance/proposal/page.tsx

```diff
@@ -0,0 +1,315 @@
+"use client";
+
+import { useRouter, useSearchParams } from "next/navigation";
+import Text from "@/components/text";
+import styles from "./proposalModal.module.scss";
+import {
+  calculateVotePercentages,
+  formatDeposit,
+  formatProposalStatus,
+  formatProposalType,
+  formatTime,
+} from "@/utils/gov/formatData";
+import Icon from "@/components/icon/icon";
+import Button from "@/components/button/button";
+import useProposals from "@/hooks/gov/useProposals";
+import { useState } from "react";
+import { Proposal } from "@/hooks/gov/interfaces/proposal";
+import useCantoSigner from "@/hooks/helpers/useCantoSigner";
+import Splash from "@/components/splash/splash";
+import { VoteOption } from "@/hooks/gov/interfaces/voteOptions";
+import Image from "next/image";
+import { NEW_ERROR } from "@/config/interfaces";
+
+import { VotingInfoBox } from "../components/VotingInfoBox/VotingInfoBox";
+import { TransactionFlowType } from "@/transactions/flows/flowMap";
+import { NewTransactionFlow } from "@/transactions/flows/types";
+import Spacer from "@/components/layout/spacer";
+
+export default function Page() {
+  function castVote(proposalId: number, voteOption: VoteOption | null) {
+    if (signer) {
+      if (!voteOption) {
+        return NEW_ERROR("Please select a vote option");
+      }
+      //console.log("cast vote test");
+      const newFlow: NewTransactionFlow = {
+        icon: "/canto.svg",
+        txType: TransactionFlowType.VOTE_TX,
+        title: "Voting",
+        params: {
+          chainId: chainId,
+          ethAccount: signer?.account.address ?? "",
+          proposalId: proposalId,
+          voteOption: voteOption,
+        },
+      };
+      txStore?.addNewFlow({
+        txFlow: newFlow,
+        ethAccount: signer.account.address,
+      });
+    }
+  }
+
+  const searchParams = useSearchParams();
+  const id = searchParams.get("id");
+  const proposalId = Number(id);
+  const router = useRouter();
+
+  const { txStore, signer, chainId } = useCantoSigner();
+  const { proposals, isProposalsLoading } = useProposals({ chainId: chainId });
+  const [selectedVote, setSelectedVote] = useState<VoteOption | null>(null);
+
+  const [isChecked, setChecked] = useState(false);
+
+  const handleCheck = (voteOption: VoteOption) => {
+    setSelectedVote(voteOption);
+    setChecked(!isChecked);
+  };
+
+  //console.log(selectedVote);
+  if (isProposalsLoading) {
+    return <Splash />;
+  }
+
+  if (!id) {
+    return (
+      <div>
+        <Text font="proto_mono">Proposal ID is missing</Text>
+      </div>
+    );
+  }
+
+  function getVotingBoxStyles() {
```

Comment:
this should be done at the component itself. The page should not have to do this work

Reply:
Nope. This styling is not for the VotingBox component. It is for the div( or container ) which has all the 4 Voting Boxes and the Button below it. As we shouldn't render the Vote button when the Proposal isn't active and the buttons need to occupy the whole container when not active, I had done it this way. I don't think this needs to be changed